.github/workflows/lint-and-test.yaml

```diff
@@ -28,16 +40,171 @@ concurrency:
+
+      - name: Confirm decompressed
+        run: |
+          echo "Current working directory: $(pwd)"
+          echo "Contents of repository root:"
+          ls -l
+          echo "Contents of decompressed folder ($VSOMEIP_SOURCE_PATH):"
+          ls -l $VSOMEIP_SOURCE_PATH
+
+      - name: Generate cache key
+        run: echo "CACHE_KEY=${{ runner.os }}-vsomeip-${{ hashFiles(env.VSOMEIP_SOURCE_PATH_WITH_WILDCARD) }}" >> $GITHUB_ENV
+
+      - name: Set cache key output
+        id: generate_cache_key
+        run: echo "CACHE_KEY=${{ env.CACHE_KEY }}" >> $GITHUB_OUTPUT
+
+      - name: Cache vsomeip build
+        id: cache-vsomeip
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.VSOMEIP_INSTALL_PATH }}
+          key: ${{ runner.os }}-vsomeip-${{ hashFiles(env.VSOMEIP_SOURCE_PATH_WITH_WILDCARD) }}
+          restore-keys: |
+            ${{ runner.os }}-vsomeip-
+
+      - name: Install dependencies
+        if: steps.cache-vsomeip.outputs.cache-hit != 'true'
+        run: sudo apt-get install -y build-essential cmake libboost-all-dev
+
+      - name: Build vsomeip
+        if: steps.cache-vsomeip.outputs.cache-hit != 'true'
+        run: |
+          cd $VSOMEIP_SOURCE_PATH
+          cmake -Bbuild -DCMAKE_INSTALL_PREFIX=$VSOMEIP_INSTALL_PATH -DENABLE_SIGNAL_HANDLING=1 .
+          cmake --build build --target install
+
+      - name: Save vsomeip build to cache
+        if: steps.cache-vsomeip.outputs.cache-hit != 'true'
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.VSOMEIP_INSTALL_PATH }}
+          key: ${{ runner.os }}-vsomeip-${{ hashFiles(env.VSOMEIP_SOURCE_PATH_WITH_WILDCARD) }}
+
+  foo:
+    needs: download_and_decompress
+    runs-on: ubuntu-latest
+    env:
+      CACHE_KEY: ${{ needs.download_and_decompress.outputs.cache_key }}
+    steps:
+      - name: Set environment variables
+        run: |
+          echo "VSOMEIP_INSTALL_PATH=${{ github.workspace }}/${{ env.VSOMEIP_INSTALL_PATH }}" >> $GITHUB_ENV
+
+      - name: Print environment variables for debugging
+        run: |
+          echo "VSOMEIP_INSTALL_PATH: ${{ env.VSOMEIP_INSTALL_PATH }}"
+
+      - name: Ensure vsomeip install path exists
+        run: |
+          mkdir -p ${{ env.VSOMEIP_INSTALL_PATH }}
+          ls -l
+
+      - name: Print cache_key
+        run: |
+          echo "cache_key: ${{ env.CACHE_KEY }}"
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Restore vsomeip build cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.VSOMEIP_INSTALL_PATH }}
+          key: ${{ env.CACHE_KEY }}
+          restore-keys: |
+            ${{ runner.os }}-vsomeip-
+
+      - name: Use vsomeip installation
+        run: |
+          echo "VSOMEIP install path from previous job: ${{ env.VSOMEIP_INSTALL_PATH }}"
+          ls -l ${{ env.VSOMEIP_INSTALL_PATH }}
+
+# TODO: Temporarily commenting this proven bit out to get the download & compilation
```

Comment:
Can be removed now?

Reply:
Again... haha, feel free to review everything but the CI :)

Reply:
Fixed